packages/oc-template-handlebars-compiler/test/compile.test.js

```diff
@@ -1,42 +1,125 @@
+);
+
+const package3 = _.cloneDeep(componentPackage);
+const package2 = _.cloneDeep(componentPackage);
+const package4 = _.cloneDeep(componentPackage);
+delete package2.oc.files.static;
+package3.oc.files.static = 'assets';
+delete package4.oc.files.data;
+
+const scenarios = {
+  'Happy path': {
     componentPackage,
     ocPackage: {
       version: '1.0.0'
     },
     componentPath,
-    publishPath
-  };
-  fs.ensureDirSync(publishPath);
-  compile(options, (err, res) => {
-    expect(err).toBeNull();
-    const version = res.oc.files.template.version;
-    res.oc.files.template.version = '';
-    res.oc.date = '';
-    expect(res).toMatchSnapshot();
-    expect(version).toBe(getInfo().version);
-    nodeDir.paths(publishPath, (err, res) => {
-      const files = res.files
+    publishPath: path.join(componentPath, '_compile-tests-package')
+  },
+  'Should handle empty static folder': {
+    componentPackage: package2,
+    ocPackage: {
+      version: '1.0.0'
+    },
+    componentPath,
+    publishPath: path.join(componentPath, '_compile-tests-package2')
+  },
+  'Should normalise stringified static folder': {
+    componentPackage: package3,
+    ocPackage: {
+      version: '1.0.0'
+    },
+    componentPath,
+    publishPath: path.join(componentPath, '_compile-tests-package3')
+  },
+  'Should handle server.js-less components': {
+    componentPackage: package4,
+    ocPackage: {
+      version: '1.0.0'
+    },
+    componentPath,
+    publishPath: path.join(componentPath, '_compile-tests-package4')
+  }
+};
+
+const execute = (options, cb) => {
+  compile(options, (err, result) => {
+    if (err) {
+      return fs.remove(options.publishPath, () => cb(err));
+    }
+
+    result.oc.date = '';
+    result.oc.files.template.version = '';
+    nodeDir.paths(options.publishPath, (err2, res2) => {
+      const files = res2.files
         .map(filePath => path.relative(__dirname, filePath))
         .sort();
+      fs.remove(options.publishPath, () => cb(err, { result, files }));
+    });
+  });
+};
+
+_.each(scenarios, (scenario, testName) => {
+  test(testName, done => {
+    execute(scenario, (err, { result, files }) => {
+      expect(err).toBeNull();
+      expect(result).toMatchSnapshot();
       expect(files).toMatchSnapshot();
-      fs.emptyDirSync(publishPath);
       done();
     });
   });
 });
+
+test('When server compilation fails should return an error', done => {
+  let package5 = _.cloneDeep(componentPackage);
+  package5.oc.files.data = 'not-found.js';
+
+  const options = {
+    componentPackage: package5,
+    ocPackage: {
+      version: '1.0.0'
+    },
+    componentPath,
+    publishPath: path.join(componentPath, '_compile-tests-package5')
+  };
+
+  execute(options, err => {
+    expect(err).toContain('Entry module not found');
+    done();
+  });
+});
+
+test('When files writing fails should return an error', done => {
+  const original = fs.ensureDir;
+  fs.ensureDir = jest.fn((a, cb) => cb('sorry I failed'));
```

Comment:
Instead of overriding and then resetting the "original" method of fs-extra we could use the jest [automocking feat](https://facebook.github.io/jest/docs/jest-object.html#jestgenmockfrommodulemodulename)?